pandas/parser.pyx

```diff
@@ -482,11 +485,14 @@ cdef class TextReader:
 
         self.verbose = verbose
         self.low_memory = low_memory
-        self.parser.converter = xstrtod
+        self.parser.double_converter = xstrtod
+        self.parser.double_converter_nogil = True
```

Comment:
instead of an indicator variable, why don't we do this

```
self.parser.double_converter_nogil = xstrtod
self.parser.double_converter_withgil = None
if float_precision == ..: 
....
elif float_precsion == 'round_trip':
    self.parser.double_converter_nogil = None
    self.parser.double_converter_withgil = round_trip
```

Then the function pointer itself is descriptive. same effect, just easier to read (and you don't have to know what the boolean is).